### DISCUSSION TITLE
Seeking Scientific Articles on SNARIMAX Model in Incremental Approach

### DISCUSSION THREAD

**Discussion by jpolnasc** [online-ml/river#1322] [General]
Hello fellow developers and researchers,

I hope this message finds you well. I've been delving into the SNARIMAX model within the incremental approach provided by this Python library, with a keen interest in its scientific applications and theoretical underpinnings. However, my quest to find scholarly articles or academic papers elucidating this topic has proven quite challenging.

I've conducted extensive searches in academic databases and journals, but I've come up empty-handed in my pursuit of scientific articles that delve into the SNARIMAX model's incremental implementation within the context of this library.

If any of you are aware of, or have come across, scientific articles or academic papers that discuss the SNARIMAX model within this library's incremental framework, I would be immensely grateful if you could share the references or links with me. It's crucial for my research to access credible and peer-reviewed sources that provide insights into the model's mathematical foundations, applications, and empirical studies.

Your assistance in locating such scientific resources would be of great value to me and, I believe, to the broader community interested in this subject matter. Thank you for your time and expertise!

**Comment by MaxHalford** [comment]
Hey there @jpolnasc. Have you checked the links at the bottom of [this](https://riverml.xyz/0.19.0/api/time-series/SNARIMAX/) page in the docs?

**Reply by jpolnasc** [reply]
Yes, I did check the links at the bottom of this page in the docs. They cover the theoretical convergence of incremental ARMA, which was helpful. However, I couldn't find specific references to ARIMA or SARIMA there. If you happen to know of any additional resources or have any information on these topics, I would greatly appreciate it! Thanks for your help nonetheless.

**Reply by MaxHalford** [reply]
Right.

The I part of ARIMA is just the regular integration step. It's just an extra step you can add on top of ARMA. In River this is done via the [`Differencer` class](https://github.com/online-ml/river/blob/main/river/time_series/snarimax.py#L12-L95), wherein you can find a couple of academic references.

The S part is a further enhancement of the I part: differencing can be done in a seasonal manner. Again, the references in the `Differencer` class apply.

The N part refers to the fact any model can be used to fit the weights of the model. Usually a linear Kalman filter used, and that's the typical way ARIMA is done.

The X part is because extra features can be passed to the model. Usually, ARIMA only learns on the lags of the target variable. But nothing is stopping you from providing extra features.

I hope that helps! I'm sorry but I don't have any other academic references to provide. The SNARIMAX class in River is mostly a patchwork of ideas I gleaned from here and there. A graduate student from Japan called Shogo Tanaka also spotted many little mistakes (see [here](https://github.com/online-ml/river/pull/980), [here](https://github.com/online-ml/river/pull/988), and [here](https://github.com/online-ml/river/pull/992)).
